packages/mdc-list/foundation.js

```diff
@@ -109,29 +112,36 @@ class MDCListFoundation extends MDCFoundation {
 
   /** @param {number} index */
   setSelectedIndex(index) {
-    if (index === this.selectedIndex_) {
+    if (index < 0 && index >= this.adapter_.getListItemCount()) {
       return;
     }
 
-    const className = this.useActivatedClass_
-      ? cssClasses.LIST_ITEM_ACTIVATED_CLASS : cssClasses.LIST_ITEM_SELECTED_CLASS;
+    const prevSelectedIndex = this.selectedIndex_;
+    const hasCheckboxOrRadio = this.adapter_.hasCheckboxOrRadioAtIndex(index);
+    const ariaAttribute = hasCheckboxOrRadio ? strings.ARIA_CHECKED : strings.ARIA_SELECTED;
 
-    if (this.selectedIndex_ >= 0) {
-      this.adapter_.removeAttributeForElementIndex(this.selectedIndex_, strings.ARIA_SELECTED);
-      this.adapter_.removeClassForElementIndex(this.selectedIndex_, className);
-      this.adapter_.setAttributeForElementIndex(this.selectedIndex_, 'tabindex', -1);
+    // Update aria attributes
+    const hasCheckbox = this.adapter_.hasCheckboxAtIndex(index);
+    if (!hasCheckbox) {
+      this.adapter_.setAttributeForElementIndex(prevSelectedIndex, ariaAttribute, false);
+    }
+    const ariaAttributeValue = hasCheckbox ? this.adapter_.isCheckboxCheckedAtIndex(index) : true;
+    this.adapter_.setAttributeForElementIndex(index, ariaAttribute, ariaAttributeValue ? 'true' : 'false');
+
+    // Update class names
+    const selectedClassName = this.useActivatedClass_ ?
+      cssClasses.LIST_ITEM_ACTIVATED_CLASS :
+      cssClasses.LIST_ITEM_SELECTED_CLASS;
+    if (!hasCheckboxOrRadio) {
```

Comment:
This seems like a confusing workaround for the fact that we're (incorrectly) setting single selection mode in the presence of checkbox or radio inputs now. (See my other comment in index.js)